### PR TITLE
Webvideocamera

### PR DESCRIPTION
**Pull request by thomasflynn**
Added GVRWebViewSceneObject, VideoSceneObject, and CameraSceneObjects to the scene-objects sample code.

GearVRf-DCO-1.0-Signed-off-by: Thomas Flynn 
tom.flynn@samsung.com

### CODE REVIEW

**Comment by phyuzion** [comment]
I think this is enough code for full sample. :)

**Comment by tflynnt** [comment]
I was inspired by you :-).  Your pull-request the other week was a good idea.  I just wanted it in a separate sample.  There's one last scene object i want in: GVRTextSceneObject.  That needs some more debugging before it can go in.

File: GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SceneObjectActivity.java [hunk 1]
```diff
@@ -30,15 +37,87 @@
...
-        mViewManager = new SampleViewManager();
+
+        createWebView();
+        createCameraView();
+
+        mViewManager = new SampleViewManager(this);
         setScript(mViewManager, "gvr_note4.xml");
     }
 
+    private void createWebView() {
```
Comment (jason2kim) [inline GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SceneObjectActivity.java]: I can see it reflects my previous comments on Insu’s including the UI thread and non-commercial sample url. No further comment for me at this moment.

File: GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java [hunk 2]
```diff
@@ -64,15 +75,23 @@ public void onInit(GVRContext gvrContext) {
         GVRCylinderSceneObject cylinderObject = new GVRCylinderSceneObject(
                 gvrContext);
         GVRConeSceneObject coneObject = new GVRConeSceneObject(gvrContext);
+        GVRWebViewSceneObject webViewObject = createWebViewObject(gvrContext);
+        GVRCameraSceneObject cameraObject = new GVRCameraSceneObject(
+                gvrContext, 8.0f, 4.0f,
+                ((SceneObjectActivity) mActivity).getCamera());
```
Comment (JonShemitz) [inline GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java]: Why cast here? Why not just declare `mActivity` as `SceneObjectActivity` (on line 54) and pass it as a `SceneObjectActivity` (on line 56)?

File: GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java [hunk 3]
```diff
@@ -64,15 +75,23 @@ public void onInit(GVRContext gvrContext) {
...
         objectList.add(cylinderObject);
         objectList.add(coneObject);
-        
+        objectList.add(webViewObject);
+        objectList.add(cameraObject);
+        objectList.add(videoObject);
+
         // turn all objects off, except the first one
-        for(int i=1; i<objectList.size(); i++) {
+        for (int i = 1; i < objectList.size(); i++) {
```
Comment (JonShemitz) [inline GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java]: Calling `.size()` a lot ...

File: GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java [hunk 4]
```diff
@@ -123,7 +159,12 @@ public void setXAngle(float angle) {
 
     public void onTap() {
 
-        objectList.get(currentObject).getRenderData().setRenderMask(0);
+        GVRSceneObject object = objectList.get(currentObject);
+        object.getRenderData().setRenderMask(0);
```
Comment (JonShemitz) [inline GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java]: Should we add a `GVRRenderMaskBit` value of 0?

File: GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java [hunk 5]
```diff
@@ -131,12 +172,15 @@ public void onTap() {
...
-                        GVRRenderData.GVRRenderMaskBit.Left
-                                | GVRRenderData.GVRRenderMaskBit.Right);
+        object = objectList.get(currentObject);
+        if (object instanceof GVRVideoSceneObject) {
+            GVRVideoSceneObject video = (GVRVideoSceneObject) object;
+            video.getMediaPlayer().start();
+        }
+        
+        object.getRenderData().setRenderMask(
+                GVRRenderData.GVRRenderMaskBit.Left
```
Comment (JonShemitz) [inline GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java]: We probably should add a `Both` constant ...

File: GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SceneObjectActivity.java [hunk 6]
```diff
@@ -30,15 +37,87 @@
...
+            @Override
+            public boolean shouldOverrideUrlLoading(WebView view, String url) {
+                view.loadUrl(url);
+                return true;
+            }
+        });
+    }
+
+    private boolean checkCameraHardware(Context context) {
+        if (context.getPackageManager().hasSystemFeature(
```
Comment (JonShemitz) [inline GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SceneObjectActivity.java]: What's wrong with a simple 

    return context.getPackageManager().hasSystemFeature(PackageManager.FEATURE_CAMERA);

?

**Comment by JonShemitz** [comment]
Looks OK - I don't see anything wrong, just a very few things that I'd do differently.

File: GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java (hunk 2 above)
Comment (thomasflynn) [inline GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java]: It was late at night.  What you suggest is certainly much cleaner.  Just made that change in my latest patchset.

File: GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java (hunk 3 above)
Comment (thomasflynn) [inline GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java]: well, like 8 times or so.  still, fixed it in my latest patchset.

File: GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java (hunk 5 above)
Comment (thomasflynn) [inline GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SampleViewManager.java]: Agreed.  we should do that in a separate check-in for GVRRenderData.  We should add a 'both' and 'none' or 'off' or something.

File: GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SceneObjectActivity.java (hunk 6 above)
Comment (thomasflynn) [inline GVRf/Sample/scene-objects/src/org/gearvrf/sample/sceneobjects/SceneObjectActivity.java]: heh, it's what you get when you read the google documentation on the camera :)  http://developer.android.com/guide/topics/media/camera.html

I changed it to the simple return you suggest in my latest commit.

**Comment by JonShemitz** [comment]
Am I missing something? I pulled the commit to test it, but can't import it into Eclipse because there doesn't seem to be a `.project` file.

**Comment by thomasflynn** [comment]
i can add that .project file.  is that necessary?  i don't remember having to do that before.

**Comment by JonShemitz** [comment]
> i can add that .project file. is that necessary? i don't remember having to do that before.

Yes, Eclipse won't import a project without the `.project` file. You probably don't remember doing so either because it's been a while since you've added a new project to a repository or just because it's the sort of fussy little detail that it's easy to forget.

**Comment by thomasflynn** [comment]
ok, added the project file.

**Comment by JonShemitz** [comment]
> ok, added the project file.

Woops - no `.classpath`, either.

**Comment by thomasflynn** [comment]
ok done.

**Comment by thomasflynn** [comment]
any further requests on this, guys?

**Comment by JonShemitz** [comment]
Does it work, Chao?

**Comment by chenchao1407** [comment]
Tried. Everything seems good. Only a small concern on video. Once it got stopped, it will never come back to play state. I took a look at the codes to see if it is based on design or a bug.  I think it is latter.

**Comment by thomasflynn** [comment]
ok, paused the video rather than stopping it.

**Pull request merged**